ocean_lib/common/ddo/ddo.py

```diff
@@ -354,3 +387,64 @@ def create_authentication_from_json(values):
 
         authentication = {"type": authentication_type, "publicKey": key_id}
         return authentication
+
+    def get_addresses_of_type(self, section_type="allow"):
```

Comment:
I wouldn't go into generalising quite yet. That is why I only left some comments indicating that there can be extensions in the future. Why?
1. this PR already does a lot of things
2. we don't know which transformations apply to other credentials. E.g. address must be lowercased for any comparison, but maybe other credentials are case sensitive or need some symbol stripping etc. Which is why we can't generalise _now_.

Ref. your point (1), I agree the naming is bad now, but I'm not in love with `allow_or_deny` either, because it contains an `or`, and it also restricts. Will all future credentials only have `allow` and `deny`, or could there be other modifiers? I'll add some clarity here, but I need to give it some more thought.

Reply:
> we don't know which transformations apply to other credentials. E.g. address must be lowercased for any comparison, but maybe other credentials are case sensitive or need some symbol stripping etc. Which is why we can't generalise now.

I totally missed this in my initial review.  You're right. It's too early to generalize.

> Will all future credentials only have allow and deny, or could there be other modifiers? 

Standard ACL rules used in networking are either "permit" or "deny". Other modifiers are treated as separate parameters. That said, I like your chosen variable name "access_class" which resolves my concern about the variable name.